compiler/rustc_infer/src/infer/type_variable.rs

```diff
@@ -357,11 +357,11 @@ impl<'tcx> TypeVariableTable<'_, 'tcx> {
         &mut self,
         value_count: usize,
     ) -> (Range<TyVid>, Vec<TypeVariableOrigin>) {
-        let range = TyVid { index: value_count as u32 }..TyVid { index: self.num_vars() as u32 };
+        let range = TyVid::from_u32(value_count as u32)..TyVid::from_usize(self.num_vars());
```

Comment:
This could be `TyVid::from_usize(value_count)` right?

Reply:
Yeah, good catch.